File: mainstream-android/app/src/main/java/com/mainstream/app/data/model/Route.kt
package com.mainstream.app.data.model

data class PredefinedRoute(
    val id: Long,
    val name: String,
    val description: String? = null,
    val city: String? = null,
    val imageUrl: String? = null,
    val originalFilename: String,
    val distanceMeters: Double,
    val elevationGainMeters: Double? = null,
    val elevationLossMeters: Double? = null,
    val startLatitude: Double? = null,
    val startLongitude: Double? = null,
    val isActive: Boolean,
    val trackPointCount: Int,
    val createdAt: String,
    val updatedAt: String,
    val trackPoints: List<RouteTrackPoint>? = null,
    val stats: RouteStats? = null
) {
    val distanceKm: Double
        get() = distanceMeters / 1000.0

    val formattedDistance: String
        get() = "%.2f km".format(distanceKm)
}

data class RouteTrackPoint(
    val id: Long,
    val sequenceNumber: Int,
    val latitude: Double,
    val longitude: Double,
    val elevation: Double? = null,
    val distanceFromStartMeters: Double
)

data class RouteStats(
    val routeId: Long,
    val todayCount: Int,
    val thisWeekCount: Int,
    val thisMonthCount: Int,
    val thisYearCount: Int,
    val totalCount: Int
)

File: mainstream-ios/Mainstream/Models/Route.swift
import Foundation

// MARK: - Predefined Route Model
struct PredefinedRoute: Codable, Identifiable {
    let id: Int
    let name: String
    let description: String?
    let city: String?
    let imageUrl: String?
    let originalFilename: String
    let distanceMeters: Double
    let elevationGainMeters: Double?
    let elevationLossMeters: Double?
    let startLatitude: Double?
    let startLongitude: Double?
    let isActive: Bool
    let trackPointCount: Int
    let createdAt: String
    let updatedAt: String
    let trackPoints: [RouteTrackPoint]?
    let stats: RouteStats?

    var distanceKm: Double {
        distanceMeters / 1000
    }

    var formattedDistance: String {
        String(format: "%.2f km", distanceKm)
    }

    var formattedElevationGain: String {
        guard let elevation = elevationGainMeters else { return "N/A" }
        return String(format: "%.0f m", elevation)
    }

    var formattedElevationLoss: String {
        guard let elevation = elevationLossMeters else { return "N/A" }
        return String(format: "%.0f m", elevation)
    }
}

// MARK: - Route Track Point
struct RouteTrackPoint: Codable, Identifiable {
    let id: Int
    let sequenceNumber: Int
    let latitude: Double
    let longitude: Double
    let elevation: Double?
    let distanceFromStartMeters: Double
}

// MARK: - Route Stats
struct RouteStats: Codable {
    let routeId: Int
    let todayCount: Int
    let thisWeekCount: Int
    let thisMonthCount: Int
    let thisYearCount: Int
    let totalCount: Int
}
